src/otx/core/types/label.py

```diff
@@ -283,16 +282,6 @@ class SegLabelInfo(LabelInfo):
 
     ignore_index: int = 255
 
-    def __post_init__(self):
```

Comment:
What's the reason that you removed this `__post_init__()`? If you remove this, it means that `SegLabelInfo` allows `label_names` even if there is no `Background` in itself. Is it mandatory to have `background` label?

Reply:
That is one of point in my PR. Remove inserting mandatory background label if we have masks as annotations. Because not all datasets require that, we can have all objects being annotated. Also, instead of background we can have like "sky", "sea", what ever. Mask annotations are consist of all labels and aren't required to insert background label. I didn't like that so much time in 1.6, but we kept it for Geti and now in 2.0 we can divide Geti case from OTX use case.

Going further, sometimes we train model for example with 3 classes in head, when we actually have 2 classes. It leads even to slightly performance change.

Reply:
I see. But, based on this comment, https://github.com/openvinotoolkit/training_extensions/pull/3515#discussion_r1604774122, I think that the real constraint from the model side exists: the number of labels should be more than one. This should be checked in `__post_init__()`

Reply:
In case of mask annotations it will be tracked by https://github.com/openvinotoolkit/training_extensions/pull/3515/files/043abd59e713391c61686b5041296e46f67ff57f#diff-4f6bc95c31fd707daea6b53a000198947e0a7b9d100c171d8a133b6615a2dc85R75
In case of polygons -> background class will be inserted

Reply:
The current implementation cannot fail if the dataset has only one label. However, the model implementation wants to fail at this case5. What I requested is that this model side restriction is realized here.